src/ros_filter.cpp

```diff
@@ -1916,6 +1916,15 @@ namespace RobotLocalization
       {
         ROS_ERROR_STREAM("Critical Error, NaNs were detected in the output state of the filter." <<
               " This was likely due to poorly conditioned process, noise, or sensor covariances.");
+        if(last_odom.header.stamp.sec > 0.0)
+        {
+          ROS_ERROR_STREAM("Apply Previous Pose");
+          geometry_msgs::PoseWithCovarianceStamped last_pose;
+          last_pose.header = last_odom.header;
+          last_pose.pose = last_odom.pose;
+          setPoseCallback(boost::make_shared<geometry_msgs::PoseWithCovarianceStamped>(last_pose));
+          // we might want to early return
+        }
```

Comment:
Maybe have an else condition that just sets state to 0?

Reply:
it's taken care of in the setposecallback

Reply:
How so?  Isn't setPoseCallback not even called if last_odom stamp is 0?

Reply:
I remember we had bad experience using setpose service (lot of instabilities), do you think we can experience the same with the callback @rbtchun3 ?

Reply:
@SalimZRai this way will not have the congestion issues as everything is running in linear sequence(I believe that's what causes instabilities of ROS as system struggles)